### PR TITLE
Describe the philosophy of VISSL benchmark dataset contribution and enrich the list of available benchmarks (take 2)

### PR DESCRIPTION
**Pull request by QuentinDuval**
This PR is a rebase of https://github.com/facebookresearch/vissl/pull/181, completed with a few additional benchmarks and improvements:

- the number datasets has increased from 2 to 9 (mostly inspired by the VTAB benchmark suite and the CLIP paper)
- data preparation scripts now all follow the same interface (same parameters to execute the script)
- data preparation scripts now all automatically download the dataset in case this one is not present
- some benchmarks have been moved out of "datasets" since their results are in line with the CLIP paper results

This PR needs to be further completed:
- Documentation has been reworked, but still needs work to add all the scripts.
- Benchmarks also need some polishing.

### CODE REVIEW

**Comment by prigoyal** [comment]
this is a LOT of really AWESOME work @QuentinDuval ! and look at all those new datasets numbers!! very exciting and the documentation is excellent. the PR is good to go. just one major comment for the full PR

we need to add the copyright `# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved` to all the `.py` files at the very top as the very first line! :)

**Review by prigoyal — APPROVED** [review]
AWESOME WORK!!!! Thank you Quentin!!

**Review by prigoyal — APPROVED** [review]
This is awesome @QuentinDuval ! I think this is good to go and merge whenever you are ready! the #datasets have gone up so much in this PR. all thanks and kudos to your hard work! 

Our next step is to import this PR in fbcode "Import to Phabricator" and ensure the tests run/pass and then we merge it!!

Thank you again!! 

2 more follow-ups separately from the PR:
- we should post in VISSL users group about the addition of the datasets and benchmark expansion
- we should expand VISSL readme to now include a bullet point on datasets as well.

File: vissl/data/README.md [hunk 1]
```diff
@@ -126,6 +126,57 @@ stl10_binary/
...
+
+VISSL supports benchmarks inspired by the [VTAB](https://arxiv.org/pdf/1910.04867.pdf) and [CLIP](https://cdn.openai.com/papers/Learning_Transferable_Visual_Models_From_Natural_Language_Supervision.pdf) papers, for which the datasets either:
+
+- Do not directly exist but are transformations of existing dataset
+- Are not in a format directly compatible with the `disk_folder` format of VISSL
+
+To run these benchmarks, the following data preparation scripts are mandatory:
+
+- `create_clevr_count_data_files.py`: to create a dataset from [CLEVR](https://cs.stanford.edu/people/jcjohns/clevr/) where the goal is to count the number of object in the scene
+- `create_clevr_dist_data_files.py`: to create a dataset from [CLEVR](https://cs.stanford.edu/people/jcjohns/clevr/) where the goal is to estimate the distance of the closest object in the scene
```
Comment (prigoyal) [inline vissl/data/README.md:139]: we should mention what the format created will be: `disk_filelist` or `disk_folder` :)

File: vissl/data/README.md [hunk 2]
```diff
@@ -126,6 +126,57 @@ stl10_binary/
...
+## Expected dataset structure for the other benchmark datasets
+
+VISSL supports benchmarks inspired by the [VTAB](https://arxiv.org/pdf/1910.04867.pdf) and [CLIP](https://cdn.openai.com/papers/Learning_Transferable_Visual_Models_From_Natural_Language_Supervision.pdf) papers, for which the datasets either:
+
+- Do not directly exist but are transformations of existing dataset
+- Are not in a format directly compatible with the `disk_folder` format of VISSL
+
+To run these benchmarks, the following data preparation scripts are mandatory:
+
+- `create_clevr_count_data_files.py`: to create a dataset from [CLEVR](https://cs.stanford.edu/people/jcjohns/clevr/) where the goal is to count the number of object in the scene
```
Comment (prigoyal) [inline vissl/data/README.md:138]: we should mention what the format created will be: `disk_filelist` or `disk_folder` :)

File: vissl/data/README.md [hunk 3]
```diff
@@ -126,6 +126,57 @@ stl10_binary/
...
+- `create_clevr_dist_data_files.py`: to create a dataset from [CLEVR](https://cs.stanford.edu/people/jcjohns/clevr/) where the goal is to estimate the distance of the closest object in the scene
+- `create_euro_sat_data_files.py`: to transform the [EUROSAT](https://github.com/phelber/eurosat) dataset to the `disk_folder` format
+- `create_food101_data_files.py`: to transform the [FOOD101](https://data.vision.ee.ethz.ch/cvl/datasets_extra/food-101) dataset to the `disk_folder` format
+- `create_patch_camelyon_data_files.py`: to transform the [PatchCamelyon](https://github.com/basveeling/pcam) dataset to the `disk_folder` format
+- `create_small_norb_azimuth_data_files.py` to create a dataset from [Small NORB](https://cs.nyu.edu/~ylclab/data/norb-v1.0-small/) where the goal is to find the azimuth or the photographed object
```
Comment (prigoyal) [inline vissl/data/README.md:143]: we should mention what the format created will be: `disk_filelist` or `disk_folder` :)

File: vissl/data/README.md [hunk 4]
```diff
@@ -126,6 +126,57 @@ stl10_binary/
...
+- `create_small_norb_elevation_data_files.py` to create a dataset from [Small NORB](https://cs.nyu.edu/~ylclab/data/norb-v1.0-small/) where the goal is to predict the elevation in the image
```
Comment (prigoyal) [inline vissl/data/README.md:144]: we should mention what the format created will be: `disk_filelist` or `disk_folder` :)

Comment (QuentinDuval) [reply vissl/data/README.md:144]: Indeed :) It should be fixed at f618c50.

File: vissl/data/README.md (hunk 3 above)
Comment (QuentinDuval) [reply vissl/data/README.md:143]: Indeed :) It should be fixed at f618c50.

File: vissl/data/README.md (hunk 2 above)
Comment (QuentinDuval) [reply vissl/data/README.md:138]: Indeed :) It should be fixed at f618c50.

File: vissl/data/README.md (hunk 1 above)
Comment (QuentinDuval) [reply vissl/data/README.md:139]: Indeed :) It should be fixed at f618c50.

**Comment by prigoyal** [comment]
@QuentinDuval , it seems like we need to rebase the PR on latest vissl master (one gpu test that's failing is a flaky test but fixed in vissl master) :) We should also click on "Import to Phabricator" again (the import failed due to old master) :)

**Comment by prigoyal** [comment]
It looks like we figured out internally what was going on :) The next steps are:
1. You should commandeer your diff back . To do so, go to the phabricator and scroll all the way down to the "Comment" box and you will see the option Commandeer option when you click on Comment :) 
2. Fix the linter errors -> we should likely do that in a follow-up PR and aim to land this PR as is without any further shipIT issues

**Pull request closed without merging**